ldcbench.api/src/main/java/org/dice_research/ldcbench/graph/GrphBasedGraph.java

```diff
@@ -253,4 +263,16 @@ public void setGraphIdOfNode(int nodeId, int graphId, int externalId) {
     public String toString() {
         return graph.toString();
     }
+
+    @Override
+    public void addBlankNodes(int nodeCount) {
+        blankNodesIndex = this.getNumberOfNodes();
+        addNodes(nodeCount);
+        for (int i = blankNodesIndex; i < this.getNumberOfNodes(); i++) {
+            //Get a random Source Node
+            int sourceNode = (new Random()).nextInt(blankNodesIndex);
```

Comment:
Randomly choosing a source node is fine. However, the way you implemented it has two drawbacks.
1. It is not possible to easily implement another way to choose a node.
2. It is impossible to repeat a run. Each component of ORCA receives a seed value. It is able to derive other seed values from this initial value. These other seed values can be used to initialize processes that are based on random numbers like this one.